### DISCUSSION TITLE
is it possible to obtain a NES score for each cell and overlay it on umap?

### DISCUSSION THREAD

**Discussion by carlosaya2** [satijalab/seurat#4710] [General]
Hi, it is my understanding that GSEA NES is obtained by interrogating the DEGs of each cluster in MSigDB hallmarks, is it possible to obtain the NES of each hallmark per cell and overlay it on umap plots? I will highly appreciate your feedback. cheers

**Comment by timoast** [comment]
Not sure what the NES score is but if you have any numeric data per cell you can overlay it on the UMAP representation. If you add the data as metadata in the Seurat object, you can then specify that value using the `features` parameter in `FeaturePlot()`. See the documentation for `FeaturePlot()` for more

**Comment by carlosaya2** [comment]
Hi Tim,

Thank you so much for your reply.

My question, may need to reword it, is sort of divided into two: 1. can I do GSEA in each cell (normally is performed between two conditions or within a sample ssGSEA) to identify enriched pathways (hallmarks) for each cell. As output from the analysis (2) , hypothetically we could get Normalised enriched scores (NES) for each pathway/hallmark and each cell and based on your reply be able to overlay it on umap using the  Feature plot() function, which is great. However, unsure I am unsure if there is a tool to perform single cell GSEA. Any feedback/ideas will be greatly appreciated.

Cheers,

Carlos

Carlos Aya-Bonilla BSc., MSc., PhD
Jerome Lejeune Foundation Post-Doctoral Research Fellow | Leukaemia and Cancer Genetics |Telethon Kids Institute
E: ***@***.******@***.***> | P: +61 63191553 | M: +61 4 33596038
At Telethon Kids, we value and encourage flexible working. If you receive an email outside of your work hours, there is no expectation that you will reply until you are next working.
Northern Entrance, Perth Children's Hospital, 15 Hospital Avenue, Nedlands, Western Australia, 6009 PO Box 855, West Perth, Western Australia, 6872
[page1image2826223232]<http://www.telethonkids.org.au/>
Proudly supported by the people of Western Australia through Channel 7's Telethon
Telethon Kids Institute acknowledges Aboriginal and Torres Strait Islander people as the Traditional Custodians of the land and waters of Australia. We also acknowledge the Nyoongar Wadjuk, Yawuru, Kariyarra and Kaurna Elders, their people and their land upon which the Institute is located and seek their wisdom in our work to improve the health and development of all children.



From: Tim Stuart ***@***.***>
Reply to: satijalab/seurat ***@***.***>
Date: Saturday, 3 July 2021 at 12:46 am
To: satijalab/seurat ***@***.***>
Cc: Carlos Aya-Bonilla ***@***.***>, Author ***@***.***>
Subject: Re: [satijalab/seurat] is it possible to obtain a NES score for each cell and overlay it on umap? (#4710)


Not sure what the NES score is but if you have any numeric data per cell you can overlay it on the UMAP representation. If you add the data as metadata in the Seurat object, you can then specify that value using the features parameter in FeaturePlot(). See the documentation for FeaturePlot() for more

**Reply by timoast** [reply]
Maybe what you're looking for is more of a module score: https://www.rdocumentation.org/packages/Seurat/versions/4.0.3/topics/AddModuleScore

**Comment by rachael-z** [comment]
@carlosaya2 
Hey Carlos, perhaps a package such as "escape" is what you're looking for?
http://www.bioconductor.org/packages/release/bioc/vignettes/escape/inst/doc/vignette.html

Cheers,
Rachael Zemek
